blood/src/main/java/cz/cuni/mff/d3s/blood/report/Report.java

```diff
@@ -0,0 +1,59 @@
+package cz.cuni.mff.d3s.blood.report;
+
+import cz.cuni.mff.d3s.blood.report.dump.DumpRegistration;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class Report {
+    private static Report instance = null;
+    private static System.Logger LOGGER = System.getLogger(Report.class.getName());
+    List<DumpRegistration> registrations = new ArrayList<>();
+
+    private Report() {
+        Runtime.getRuntime().addShutdownHook(new Thread(this::onShutdown, "Report shutdown hook"));
+    }
+
+    public static Report getInstance() {
+        if (instance == null)
+            instance = new Report();
+
+        return instance;
+    }
+
+    public void registerDump(DumpRegistration registration) {
+        registrations.add(registration);
+    }
+
+    private void onShutdown() {
+        var startTime = Instant.now();
+
+        new File(DumpHelpers.DUMPS_DIR_NAME).mkdir();
```

Comment:
Already in `DumpHelpers::getDumpFileWriter`. I suggest removing it from there and leaving it here.

Reply:
Ok. Will be fixed.